### PR TITLE
FEATURE: TNT-1185 Support dashboard permissions in SDK-backends

### PR DESCRIPTION
**Pull request by tomassatinsky**
JIRA: TNT-1185

---

Supported PR commands:

| Command                                               | Description                                                |
| ----------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                          | Re-run standard checks                                     |
| `extended check sonar`                                | SonarQube tests                                            |
| `extended check plugins`                              | Dashboard plugins tests                                    |
| `extended test - backstop`                            | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                |                                                            |
| `extended test - tiger-cypress - isolated <testName>` | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`   | Create a new recording for isolated Tiger tests.           |
| **E2E Cypress tests commands - BEAR**                 |                                                            |
| `extended test - cypress - isolated <testName>`       | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`         | Create a new recording for isolated Bear tests.            |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11136/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13237/ : FAILURE in 7m 23s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11137/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13238/ : SUCCESS in 20m 04s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11139/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13241/ : SUCCESS in 20m 36s

File: libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/button/selectors.ts [hunk 1]
```diff
@@ -85,9 +85,9 @@ export function selectCanSaveDashboard(state: DashboardState) {
 export const selectHasPermissionsForShare = createSelector(
     selectEnableAnalyticalDashboardPermissions,
     selectSupportsAccessControlCapability,
-    selectCanManageACL,
-    (arePermissionsEnabled, supportsAccessControl, canManageACL) =>
-        arePermissionsEnabled && supportsAccessControl && canManageACL,
+    selectCanShareDashboardPermission,
```
Comment (tomassatinsky) [inline libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/button/selectors.ts:88]: Moving from bear-specific workspace-level permission to backend agnostic dashboard permissions.
They use the same logic on Bear - see `canShareDashboard: workspacePermissions.canManageACL`

File: libs/sdk-backend-bear/src/backend/workspace/dashboards/index.ts [hunk 2]
```diff
@@ -1012,6 +1016,39 @@ export class BearWorkspaceDashboards implements IWorkspaceDashboardsService {
...
+            const workspacePermissions = await this.authCall(() =>
+                this.permissions.getPermissionsForCurrentUser(),
+            );
+            return {
+                canEditDashboard: workspacePermissions.canManageAnalyticalDashboard,
+                canShareDashboard: workspacePermissions.canManageACL,
+                canViewDashboard: true,
+            };
+        } catch (_e) {
+            return {
```
Comment (ivanmjartan) [inline libs/sdk-backend-bear/src/backend/workspace/dashboards/index.ts:1044]: I will add console log warning that default set of permission is used.

File: libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/initializeDashboardHandler/index.ts [hunk 3]
```diff
@@ -269,6 +275,11 @@ function* initializeNewDashboard(
             }),
             uiActions.setMenuButtonItemsVisibility(config.menuButtonItemsVisibility),
             renderModeActions.setRenderMode(config.initialRenderMode),
+            dashboardPermissionsActions.setDashboardPermissions({
```
Comment (tomassatinsky) [inline libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/initializeDashboardHandler/index.ts:278]: @ivanmjartan I've added this for new dashboard. So for the author we assume he will have all the permissions for new his dashboard. Once the dashboard is saved the author should get all the permissions for real in the backend- in metadata.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3398

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/261/ : FAILURE in 1m 59s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3527/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2376/ : SUCCESS in 15m 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13252/ : SUCCESS in 20m 43s

**Pull request merged**